app/src/main/java/at/roteskreuz/stopcorona/model/repositories/InfectionMessengerRepository.kt

```diff
@@ -126,60 +141,68 @@ class InfectionMessengerRepositoryImpl(
         }
 
         downloadMessagesStateObserver.loading()
         withContext(coroutineContext) {
             try {
-                val addressPrefix = cryptoRepository.publicKeyPrefix
-                var messages: List<ApiInfectionMessage>
-                do {
-                    // downloading next 100 messages
-                    messages = apiInteractor.getInfectionMessages(addressPrefix, lastMessageId).infectionMessages
-                        ?: throw Exception("Message list is null")
-
-                    // update pointer
-                    val lastId = messages.maxBy { it.id }?.id
-                    if (lastId == null) {
-                        messages = listOf() // to stop loop
-                        continue
-                    }
-                    lastMessageId = lastId
-
-                    // try to decrypt them
-                    messages.forEach { message ->
-                        val decryptedMessage = cryptoRepository.decrypt(message.message)
-
-                        // store decrypted messages to DB
-                        if (decryptedMessage != null) {
-                            val infectionMessageContent = InfectionMessageContent(decryptedMessage)
-                            if (infectionMessageContent != null && infectionMessageContent.timeStamp.isInTheFuture().not()) {
-                                val dbMessage = infectionMessageContent.asReceivedDbEntity()
-                                infectionMessageDao.insertOrUpdateInfectionMessage(dbMessage)
-                                when (val messageType = infectionMessageContent.messageType) {
-                                    is MessageType.InfectionLevel -> {
-                                        notificationsRepository.displayInfectionNotification(messageType)
-                                        quarantineRepository.receivedWarning(messageType.warningType)
-
-                                        if (infectionMessageDao.hasReceivedRedInfectionMessages().not()) {
-                                            quarantineRepository.revokeLastRedContactDate()
-                                        }
-                                    }
-                                    MessageType.Revoke.Suspicion -> {
-                                        setSomeoneHasRecovered()
-                                        notificationsRepository.displaySomeoneHasRecoveredNotification()
-                                        databaseCleanupManager.removeReceivedGreenMessages()
-                                    }
-                                }
-                            }
-                        }
-                    }
-                } while (messages.isNotEmpty()) // repeat until we have read all messages
+                val archives :List<File> = apiInteractor.fetchBatchDiagnosisKeysBasedOnInfectionLevel(warningType)
+                val contextToken = BATCH + "," + warningType + "," + UUID.randomUUID().toString()
```

Comment:
If this format is the wanted format of the exposure notification framework, I would suggest to build the contextToken inside `ExposureNotificationRepository`.